src/graphics/internal_canvas.rs

```diff
@@ -404,6 +415,17 @@ impl<'a> InternalCanvas<'a> {
                 z: param.color.b,
                 w: param.color.a,
             },
+            // this is the actual image scale that we apply in the vertex shader.
+            // we can't apply this when we first convert the instance array drawparams because we don't know the image size yet.
```

Comment:
But... we do know `instances.image.width()` and `height()` in line 409 already, don't we?
I thought the reason you didn't use them above was that you didn't want to let the `src` field influence the scaling, but now I realize you already set that to `Rect::one`.

Btw.: is that `src(Rect::one)` necessary still? Now that the `image_scale` parameter is `None`?

Reply:
We don't know the image size at the time draw params are inserted into the instance array, we only know it now, during the draw.

I suppose forcing src to one isn't necessary anymore, yes.

Reply:
But can't we then just pass the image size as `image_scale` in line 409? (then we'd have to keep `Rect::one` I guess) instead of passing it to the shader this way?

I'm pretty sure you have a reason to do it this way, I just don't understand it yet.

Reply:
We want to apply image scaling to each instance separately in its local space. If it's applied at world space then it will scale all the vertices of the instances which is undesired (i.e., it will scale the instance positions).

Reply:
Aha! That makes sense.
Could you adapt the lower comment to say so and remove the `src(Rect::one)` above? I think then we should be done.